### PR TITLE
Fixed: SAVE MAPPING button saves empty mapping when file not uploaded(#326re7p)

### LINKED ISSUE #110: SAVE MAPPING button saves empty mapping when file not uploaded
## Current behavior

If we don not upload a file and enter some mapping name & save it, it saves in the mapping list, which is not useful.

https://user-images.githubusercontent.com/62360781/205064074-df77bc5f-bcd6-4c73-a40f-ee013d095561.mov

## Expected behavior

SAVE MAPPING button should save mapping only if 
1. File is uploaded
2. All field mappings are mapped
3. A mapping name is entered

## Can you handle fixing this bug by yourself?

- [ ] YES
- [ ] NO

## Environment details

- Browser: 
- OS: 
- Code Version:

### PR DESCRIPTION
**Pull request by shashwatbangar**
### Related Issues

Closes #110 

### Short Description and Why It's Useful

Now user can only save mapping if file is uploaded, a mapping name is provided & all fields are selected.

### Screenshots of Visual Changes before/after (If There Are Any)

https://user-images.githubusercontent.com/62360781/205568757-06defc43-3bf1-4e1c-8c56-4ad18d764c1b.mov

**IMPORTANT NOTICE** - Remember to add changelog entry

### Contribution and Currently Important Rules Acceptance

- [ ] I read and followed [contribution rules](https://github.com/hotwax/import#contribution-guideline)

### CODE REVIEW

File: src/views/PurchaseOrder.vue [hunk 1]
```diff
@@ -133,11 +133,20 @@ export default defineComponent({
         return !this.fieldMappings[id] ? id : this.generateUniqueMappingPrefId();
       },
       saveMapping() {
-        if (this.mappingName) {
+        if (this.mappingName && this.file && this.areAllFieldsSelected()) {
+          showToast(translate("Mapping saved successfully"));
```
Comment (adityasharma7) [inline src/views/PurchaseOrder.vue:137]: It should after the operation completion

File: src/views/PurchaseOrder.vue [hunk 2]
```diff
@@ -133,11 +133,20 @@ export default defineComponent({
...
           const mappingPrefId = this.generateUniqueMappingPrefId();
           this.store.dispatch('user/updateFieldMappings', { mappingPrefId, mappingPrefName: this.mappingName, mappingPrefValue: JSON.parse(JSON.stringify(this.fieldMapping)) })
         } else {
-          showToast(translate("Enter mapping name"));
+          if(!this.mappingName) {
```
Comment (adityasharma7) [inline src/views/PurchaseOrder.vue:141]: We could add these conditions at the top and return control, if they are fulfilled. The code if then doesn't need any if else

File: src/views/PurchaseOrder.vue [hunk 3]
```diff
@@ -198,6 +207,9 @@ export default defineComponent({
           this.mappingName = fieldMapping.mappingPrefName; 
         }
       },
+      areAllFieldsSelected() {
```
Comment (adityasharma7) [inline src/views/PurchaseOrder.vue:210]: We could use it for other use cases in the page

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/views/PurchaseOrder.vue (hunk 1 above)
Comment (shashwatbangar) [reply src/views/PurchaseOrder.vue]: yes sir. Fixed it

File: src/views/PurchaseOrder.vue (hunk 3 above)
Comment (shashwatbangar) [reply src/views/PurchaseOrder.vue:210]: Yes sir.

**Review by adityasharma7 — APPROVED** [review]

**Pull request merged**